addons/pos_self_order/static/src/app/models/pos_order.js

```diff
@@ -6,10 +6,13 @@ import { patch } from "@web/core/utils/patch";
 patch(PosOrder.prototype, {
     setup() {
         super.setup(...arguments);
-        this.uiState = {
-            ...this.uiState,
-            lineChanges: {},
-        };
+
+        if (!this.uiState) {
+            this.uiState = {
+                ...this.uiState,
+                lineChanges: {},
+            };
+        }
```

Comment:
May I know what's the problem here? Can you do the following?

```js
Object.assign(this.uiState, { lineChanges: {} });
```

Reply:
uiState is only added in the setup if there isn't yet an ui state coming from serialized data (indexed db)